solutions/devsprint-riccardo-1/OnboardingChallenge/Service/BundleService.swift

```diff
@@ -0,0 +1,14 @@
+//
+//  BundleService.swift
+//  OnboardingChallenge
+//
+//  Created by Strawberry Pie on 12/14/21.
+//
+
+import Foundation
+
+protocol BundleService {
+    func object(forInfoDictionaryKey: String) -> Any?
```

Comment:
As suggested a few line above, we have full control over this abstraction.
Why don't we create method that we can use easily?
For example, instead of asking for and `object`, we can ask directly for a `string(forInfoDictionaryKey: String)`. In this case, the return type can be `String` and we can implement the function easily in the extension.

Even better, we could create the abstraction we actually need. In this case, the `bundleService` could directly define an `appVersion` property whose type is `String?` and we can implement the logic in the extension.

Reply:
I reimplemented this logic